arangod/Agency/Constituent.h

```diff
@@ -147,7 +151,6 @@ class Constituent : public Thread {
   aql::QueryRegistry* _queryRegistry;
 
   term_t _term;            // term number
-  bool _cast;              // cast a vote this term
```

Comment:
Trivial note:  you removed _cast but continue to use _castLock.  How about a rename to _termLock?

Reply:
_role is also protected by _castLock, add comment?

Reply:
_leaderID too, add comment?

Reply:
Comments added, renamed to _termVoteLock.